sentry-android-okhttp/src/main/java/io/sentry/android/okhttp/SentryOkHttpInterceptor.kt

```diff
@@ -104,6 +124,115 @@ class SentryOkHttpInterceptor(
         }
     }
 
+    private fun captureEvent(request: Request, response: Response) {
+        // not possible to get a parameterized url, but we remove at least the
+        // query string and the fragment.
+        // url example: https://api.github.com/users/getsentry/repos/#fragment?query=query
+        // url will be: https://api.github.com/users/getsentry/repos/
+        // ideally we'd like a parameterized url: https://api.github.com/users/{user}/repos/
+        // but that's not possible
+        var requestUrl = request.url.toString()
+
+        val query = request.url.query
+        if (!query.isNullOrEmpty()) {
+            requestUrl = requestUrl.replace("?$query", "")
+        }
+
+        val urlFragment = request.url.fragment
+        if (!urlFragment.isNullOrEmpty()) {
+            requestUrl = requestUrl.replace("#$urlFragment", "")
+        }
+
+        if (!captureFailedRequests ||
+            !PropagationTargetsUtils.contain(failedRequestTargets, requestUrl) ||
+            !containsStatusCode(response.code)
+        ) {
+            return
+        }
+
+        val mechanism = Mechanism().apply {
+            type = "SentryOkHttpInterceptor"
+        }
+        val exception = SentryHttpClientException(
+            "Event was captured because the request status code was ${response.code}"
+        )
+        val mechanismException = ExceptionMechanismException(mechanism, exception, Thread.currentThread(), true)
+        val event = SentryEvent(mechanismException)
+
+        val hint = Hint()
+        hint.set("request", request)
+        hint.set("response", response)
+
+        // remove after fields indexed
+//        val tags = mutableMapOf<String, String>()
+//        tags["status_code"] = response.code.toString()
+//        tags["url"] = requestUrl
+
+        val sentryRequest = io.sentry.protocol.Request().apply {
+            url = requestUrl
+            // Cookie is only sent if isSendDefaultPii is enabled
+            cookies = if (hub.options.isSendDefaultPii) request.headers["Cookie"] else null
```

Comment:
Should there be some internal discussion (maybe TSC) if we actually want to add cookies?

Reply:
https://github.com/getsentry/sentry-java/pull/2287#discussion_r992479315